pysc2/agents/scripted_agent.py

```diff
@@ -49,24 +49,29 @@ def step(self, obs):
 
 class CollectMineralShards(base_agent.BaseAgent):
   """An agent specifically for solving the CollectMineralShards map."""
+  current_player = 0
 
   def step(self, obs):
     super(CollectMineralShards, self).step(obs)
-    if FUNCTIONS.Move_screen.id in obs.observation["available_actions"]:
-      player_relative = obs.observation["feature_screen"][_PLAYER_RELATIVE]
-      neutral_y, neutral_x = (player_relative == _PLAYER_NEUTRAL).nonzero()
-      player_y, player_x = (player_relative == _PLAYER_SELF).nonzero()
-      if not neutral_y.any() or not player_y.any():
+    player_units = list(filter(lambda unit: unit[1] == _PLAYER_SELF, obs.observation['feature_units']))
+    if len(player_units) == 0:
+      return FUNCTIONS.no_op()
+    player_unit = player_units[self.current_player]
+    player_xy = [player_unit[12], player_unit[13]]
```

Comment:
Can you give 12, 13, 17, etc names, even if they're just constants above the class. Eventually we should use numpy records or similar, but the current way is hard to understand. Maybe the best intermediate is an enum at the top of features.py.